retroshare-gui/src/gui/Identity/IdDialog.cpp

```diff
@@ -1560,23 +1600,16 @@ void IdDialog::loadIdentities(const std::map<RsGxsGroupId,RsGxsIdGroup>& ids_set
 		ids_set.erase(it);	// erase, so it is not considered to be a new item
 	}
 
+	ui->idTreeWidget->insertTopLevelItem(0, ownItem);
+	ui->idTreeWidget->insertTopLevelItem(0, allItem);
+	ui->idTreeWidget->insertTopLevelItem(0, contactsItem );
+
 	/* Insert new items */
 	for (std::map<RsGxsGroupId,RsGxsIdGroup>::const_iterator vit = ids_set.begin(); vit != ids_set.end(); ++vit)
 	{
 		RsGxsIdGroup data = vit->second ;
 
-	    item = NULL;
-
-	    ui->idTreeWidget->insertTopLevelItem(0, ownItem);
```

Comment:
what the hell what this doing here???

Reply:
I don't know...
3b45fd8389 (csoler              2016-04-09 17:50:46 -0400)
You want I move them on constructor?

Reply:
ah yes, good idea! So we know they are already here and we dont need to add them multiple times. Make sure there's no "ui->idTreeWidget->clear()' then. If there is, maybe it's preferable to leave it as it is in your current PR.

Reply:
Done